### ISSUE TITLE
@salesforce/core's Org.create() sets the wrong alias, causing `getConnection()` to connect to the wrong org.

### ISSUE THREAD

**Issue by jclark-dot-org** [forcedotcom/cli#2724]
**Note** the [sfdx-core repo][1]'s README directs issue reports to this repo; please let me know if this should go elsewhere.

### Summary
When passing `AuthInfo` from `StateAggregator.orgs.get()` to `Org.create()`, an incorrect alias is assigned, and a subsequent call to `getConnection` connects to the wrong org.

[1]: https://github.com/forcedotcom/sfdx-core/tree/main

### Steps To Reproduce

I've extracted a small repro:

0.  Requires a system with Salesforce CLI installed, and multiple orgs authenticated, each with an alias.
1.  Clone https://github.com/jclark-dot-org/sfcdemo.
2. `npm install`
3. `node index.js [userIdOrAlias] 2>error.log`

### Expected result
The example will use `Org.create()` to create an Org object, and `Org.getConnection()` to create a connection, then outputs the `username` and `instanceUrl` of the connection; these should match the [userIdOrAlias] from step 3 above.  Finally, it runs a simple Account query.

### Actual result
A different `username` and `instanceUrl` will be output and no Query results or errors will be output.  In the event that the `username` and `instanceUrl` values are correct, try again with another [userIdOrAlias].  This has been tested on multiple systems.

### More Detail

The command also creates an `error.log` containing more console logging; it will output the AuthInfo, Org, and Connection objects.  On inspection, you should find that AuthInfo is correct, Org is correct *except* that it has the wrong alias, and Connection is completely wrong, matching only the incorrect alias from the Org.

### System Information

zsh on Mac OS Sonoma

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.26.10",
  "nodeVersion": "node-v21.4.0",
  "osVersion": "Darwin 23.1.0",
  "rootPath": "/Users/jclark/.nvm/versions/node/v21.4.0/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.5 (core)",
    "@oclif/plugin-commands 3.1.1 (core)",
    "@oclif/plugin-help 6.0.12 (core)",
    "@oclif/plugin-not-found 3.0.9 (core)",
    "@oclif/plugin-plugins 4.1.21 (core)",
    "@oclif/plugin-search 1.0.12 (core)",
    "@oclif/plugin-update 4.1.8 (core)",
    "@oclif/plugin-version 2.0.11 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.9 (core)",
    "@oclif/plugin-which 3.0.15 (core)",
    "@salesforce/cli 2.26.10 (core)",
    "apex 3.0.20 (core)",
    "auth 3.3.3 (core)",
    "data 3.0.17 (core)",
    "deploy-retrieve 3.2.6 (core)",
    "info 3.0.18 (core)",
    "limits 3.1.5 (core)",
    "marketplace 1.0.18 (core)",
    "org 3.3.8 (core)",
    "packaging 2.1.3 (core)",
    "schema 3.1.0 (core)",
    "settings 2.0.19 (core)",
    "sobject 1.1.3 (core)",
    "source 3.1.6 (core)",
    "telemetry 3.1.10 (core)",
    "templates 56.0.11 (core)",
    "trust 3.3.2 (core)",
    "user 3.2.4 (core)"
  ]
}

```
### Additional information

Here is sample `error.log` output from my system, with some values obfuscated.  It contains the following sections:
1.  `openOrg: orgInfo:`  This is an org-info object specific to the project I extracted this from; you can see it in the repo in `OrgManager.js`  Essentially, we build a list of all orgs using `stateAggregator.orgs.readAll()`, look up the associated aliases via `stateAggregator.aliases.get()`, and maintain the info in a list to display orgs to the user.  In this example, `OrgManger.getOrgInfo()` fetches the org specified on the command line, and we log it.  `username`, `instanceUrl`, and `alias` are all correct here.  In my example, I ran the command with the alias `coreutils`, and the username is `'jason.clark@axXXXXX.com'.`
2. `authInfo:`  This is the AuthInfo object returned from `stateAggregator.orgs.read(username)`; we pass the username from the orgInfo object in step 1.  Again, `username`, `instanceUrl`, and `alias` are all correct.
3. `org:`  This is the Org object returned from `Org.create(authInfo)`.  Here we see the first problem... `username` and `instanceUrl` are correct, but `alias` is wrong; in my example, it is another authenticated org on my system with the alias `'deXXXXXX_dwqcp'`
4. `conn:`  This is the Connection object from `Org.getConnection()`.  `alias` is the same wrong value from the Org object, and now `username` (`'chXXX.XXXXX@axXXXXX.com.dwcpq'`) and `instanceUrl` match this alias, not the original alias from step 1.
5.  The query doesn't result in success or failure output; I have no idea if this is related to the above issue, but I've yet to make it work in any org.

```
(node:5537) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
openOrg: orgInfo Organization {
  accessToken: '19c5ce17XXXXXXXX',
  alias: 'coreutils',
  authStatus: 'Connected',
  clientId: 'PlatformCLI',
  created: '1706024424000',
  createdOrgInstance: 'USA434S',
  devHubUsername: 'jason.clark@axXXXXX.com',
  expirationDate: '2024-02-22',
  instanceApiVersion: '60.0',
  instanceApiVersionLastRetrieved: '2/14/2024, 7:50:59 AM',
  instanceUrl: 'https://inspiration-XXXX-XXXX-dev-ed.scratch.my.salesforce.com',
  isDevHub: false,
  isScratch: true,
  loginUrl: 'https://inspiration-XXXX-XXXX-dev-ed.scratch.my.salesforce.com/',
  orgId: '00DRL0XXXXXXXX',
  refreshToken: 'a7f074f3XXXXXXXX',
  username: 'test-iwnXXXXXXXX@example.com'
}
getConnection:  test-iwnXXXXXXXX@example.com
authInfo:  {
  accessToken: '1e916670XXXXXXXX',
  instanceUrl: 'https://inspiration-XXXX-XXXX-dev-ed.scratch.my.salesforce.com',
  orgId: '00DRL0XXXXXXXX',
  username: 'test-iwnXXXXXXXX@example.com',
  loginUrl: 'https://inspiration-XXXX-XXXX-dev-ed.scratch.my.salesforce.com/',
  refreshToken: '09dbfa0bXXXXXXXX',
  clientId: 'PlatformCLI',
  isDevHub: false,
  devHubUsername: 'jason.clark@axXXXXX.com',
  created: '1706024424000',
  expirationDate: '2024-02-22',
  createdOrgInstance: 'USA434S',
  isScratch: true,
  isSandbox: false,
  instanceApiVersion: '60.0',
  instanceApiVersionLastRetrieved: '2/14/2024, 7:50:59 AM',
  tracksSource: true
}
org:  Org {
  status: 'UNKNOWN',
  options: {
    accessToken: '1e916670XXXXXXXX',
    instanceUrl: 'https://inspiration-XXXX-XXXX-dev-ed.scratch.my.salesforce.com',
    orgId: '00DRL0XXXXXXXX',
    username: 'test-iwnXXXXXXXX@example.com',
    loginUrl: 'https://inspiration-XXXX-XXXX-dev-ed.scratch.my.salesforce.com/',
    refreshToken: '09dbfa0bXXXXXXXX',
    clientId: 'PlatformCLI',
    isDevHub: false,
    devHubUsername: 'jason.clark@axXXXXX.com',
    created: '1706024424000',
    expirationDate: '2024-02-22',
    createdOrgInstance: 'USA434S',
    isScratch: true,
    isSandbox: false,
    instanceApiVersion: '60.0',
    instanceApiVersionLastRetrieved: '2/14/2024, 7:50:59 AM',
    tracksSource: true,
    aliasOrUsername: 'deXXXXXX_dwqcp'
  },
  logger: EventEmitter {
    trace: [Function: noop],
    debug: [Function: noop],
    info: [Function: noop],
    warn: [Function: LOG],
    error: [Function: LOG],
    fatal: [Function (anonymous)],
    [Symbol(pino.serializers)]: { err: [Function: errSerializer] },
    [Symbol(pino.formatters)]: {
      level: [Function: level],
      bindings: [Function: resetChildingsFormatter],
      log: undefined
    },
    [Symbol(pino.chindings)]: ',"name":"sf","name":"sf:Org"',
    [Symbol(pino.levelVal)]: 40
  },
  configAggregator: ConfigAggregator {
    envVars: {},
    globalConfig: Config {
      contents: LWWMap {},
      statics: [Function],
      options: [Object],
      hasRead: true,
      logger: [Logger],
      path: '/Users/jclark/.sf/config.json',
      sfdxPath: '/Users/jclark/.sfdx/sfdx-config.json'
    },
    allowedProperties: [
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object]
    ]
  },
  connection: <ref *1> Connection {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    version: '60.0',
    loginUrl: 'https://login.salesforce.com',
    instanceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com',
    accessToken: '00D78000XXXXXXXX',
    refreshToken: undefined,
    userInfo: undefined,
    limitInfo: {},
    oauth2: OAuth2 {
      loginUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com',
      authzServiceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com/services/oauth2/authorize',
      tokenServiceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com/services/oauth2/token',
      revokeServiceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com/services/oauth2/revoke',
      clientId: 'PlatformCLI',
      clientSecret: undefined,
      redirectUri: 'http://localhost:1717/OauthRedirect',
      codeVerifier: undefined,
      _transport: Transport {}
    },
    sobjects: {},
    cache: Cache { _entries: [Object] },
    _callOptions: { client: 'sfdx toolbelt:' },
    _maxRequest: 10,
    _logger: Logger { _moduleName: 'connection', _logLevel: 6 },
    _logLevel: undefined,
    _transport: Transport {},
    _sessionType: 'oauth2',
    _refreshDelegate: SessionRefreshDelegate {
      _refreshFn: [Function: bound refreshFn] AsyncFunction,
      _conn: [Circular *1],
      _logger: [Logger],
      _lastRefreshedAt: undefined,
      _refreshPromise: undefined
    },
    'describe$': [Function: $fn] { clear: [Function (anonymous)] },
    'describe$$': [Function: $fn] { clear: [Function (anonymous)] },
    describeSObject: [Function: $fn] { clear: [Function (anonymous)] },
    'describeSObject$': [Function: $fn] { clear: [Function (anonymous)] },
    'describeSObject$$': [Function: $fn] { clear: [Function (anonymous)] },
    'describeGlobal$': [Function: $fn] { clear: [Function (anonymous)] },
    'describeGlobal$$': [Function: $fn] { clear: [Function (anonymous)] },
    insert: [AsyncFunction: create],
    delete: [AsyncFunction: destroy],
    del: [AsyncFunction: destroy],
    process: Process { rule: [ProcessRule], approval: [ApprovalProcess] },
    describe: [Function: $fn] { clear: [Function (anonymous)] },
    describeGlobal: [Function: $fn] { clear: [Function (anonymous)] },
    analytics: [Getter],
    apex: [Getter],
    bulk: [Getter],
    bulk2: [Getter],
    chatter: [Getter],
    metadata: [Getter],
    soap: [Getter],
    streaming: [Getter],
    tooling: [Getter],
    hasResolved: false,
    options: { authInfo: [AuthInfo], connectionOptions: [Object] },
    username: 'chXXX.XXXXX@axXXXXX.com.dwcpq',
    logger: Logger { pinoLogger: [EventEmitter], memoryLogger: undefined },
    maxApiVersion: '60.0',
    [Symbol(shapeMode)]: false,
    [Symbol(kCapture)]: false
  },
  orgId: '00D780000008hreEAA'
}
conn:  <ref *1> Connection {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  version: '36.0',
  loginUrl: 'https://login.salesforce.com',
  instanceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com',
  accessToken: '00D78000XXXXXXXX',
  refreshToken: undefined,
  userInfo: undefined,
  limitInfo: {},
  oauth2: OAuth2 {
    loginUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com',
    authzServiceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com/services/oauth2/authorize',
    tokenServiceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com/services/oauth2/token',
    revokeServiceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com/services/oauth2/revoke',
    clientId: 'PlatformCLI',
    clientSecret: undefined,
    redirectUri: 'http://localhost:1717/OauthRedirect',
    codeVerifier: undefined,
    _transport: Transport {}
  },
  sobjects: {},
  cache: Cache { _entries: { describeGlobal: [CacheEntry] } },
  _callOptions: { client: 'sfdx toolbelt:' },
  _maxRequest: 10,
  _logger: Logger { _moduleName: 'connection', _logLevel: 6 },
  _logLevel: undefined,
  _transport: Transport {},
  _sessionType: 'oauth2',
  _refreshDelegate: SessionRefreshDelegate {
    _refreshFn: [Function: bound refreshFn] AsyncFunction,
    _conn: [Circular *1],
    _logger: Logger { _moduleName: 'session-refresh-delegate', _logLevel: 6 },
    _lastRefreshedAt: undefined,
    _refreshPromise: undefined
  },
  'describe$': [Function: $fn] { clear: [Function (anonymous)] },
  'describe$$': [Function: $fn] { clear: [Function (anonymous)] },
  describeSObject: [Function: $fn] { clear: [Function (anonymous)] },
  'describeSObject$': [Function: $fn] { clear: [Function (anonymous)] },
  'describeSObject$$': [Function: $fn] { clear: [Function (anonymous)] },
  'describeGlobal$': [Function: $fn] { clear: [Function (anonymous)] },
  'describeGlobal$$': [Function: $fn] { clear: [Function (anonymous)] },
  insert: [AsyncFunction: create],
  delete: [AsyncFunction: destroy],
  del: [AsyncFunction: destroy],
  process: Process {
    rule: ProcessRule { _conn: [Circular *1] },
    approval: ApprovalProcess { _conn: [Circular *1] }
  },
  describe: [Function: $fn] { clear: [Function (anonymous)] },
  describeGlobal: [Function: $fn] { clear: [Function (anonymous)] },
  analytics: [Getter],
  apex: [Getter],
  bulk: [Getter],
  bulk2: [Getter],
  chatter: [Getter],
  metadata: [Getter],
  soap: [Getter],
  streaming: [Getter],
  tooling: [Getter],
  hasResolved: false,
  options: {
    authInfo: AuthInfo {
      usingAccessToken: false,
      options: [Object],
      stateAggregator: [StateAggregator],
      username: 'chXXX.XXXXX@axXXXXX.com.dwcpq',
      logger: [Logger]
    },
    connectionOptions: {
      version: undefined,
      callOptions: [Object],
      oauth2: [Object],
      accessToken: '00D78000XXXXXXXX',
      instanceUrl: 'https://deXXXXXXXX--dwcpq.sandbox.my.salesforce.com',
      refreshFn: [Function: bound refreshFn] AsyncFunction
    }
  },
  username: 'chXXX.XXXXX@axXXXXX.com.dwcpq',
  logger: Logger {
    pinoLogger: EventEmitter {
      trace: [Function: noop],
      debug: [Function: noop],
      info: [Function: noop],
      warn: [Function: LOG],
      error: [Function: LOG],
      fatal: [Function (anonymous)],
      [Symbol(pino.serializers)]: [Object],
      [Symbol(pino.formatters)]: [Object],
      [Symbol(pino.chindings)]: ',"name":"sf","name":"sf:connection"',
      [Symbol(pino.levelVal)]: 40
    },
    memoryLogger: undefined
  },
  maxApiVersion: '60.0',
  [Symbol(shapeMode)]: false,
  [Symbol(kCapture)]: false
}
```

**Comment by jclark-dot-org** [comment]
As requested, I have updated `sf`; output below.  The issue remains.  However, this is really a bug report for `sfdx-core`; as you can see in the [package.json of the linked repo][1] I am using `"@salesforce/core": "^6.5.3"`, the current version.

```
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.28.6",
  "nodeVersion": "node-v21.4.0",
  "osVersion": "Darwin 23.1.0",
  "rootPath": "/Users/jclark/.nvm/versions/node/v21.4.0/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.7 (core)",
    "@oclif/plugin-commands 3.1.2 (core)",
    "@oclif/plugin-help 6.0.12 (core)",
    "@oclif/plugin-not-found 3.0.10 (core)",
    "@oclif/plugin-plugins 4.2.1 (core)",
    "@oclif/plugin-search 1.0.13 (core)",
    "@oclif/plugin-update 4.1.10 (core)",
    "@oclif/plugin-version 2.0.12 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.10 (core)",
    "@oclif/plugin-which 3.0.15 (core)",
    "@salesforce/cli 2.28.6 (core)",
    "apex 3.0.22 (core)",
    "auth 3.3.11 (core)",
    "data 3.0.19 (core)",
    "deploy-retrieve 3.2.11 (core)",
    "info 3.0.21 (core)",
    "limits 3.1.7 (core)",
    "marketplace 1.0.20 (core)",
    "org 3.3.11 (core)",
    "packaging 2.1.6 (core)",
    "schema 3.1.1 (core)",
    "settings 2.0.22 (core)",
    "sobject 1.1.9 (core)",
    "source 3.1.10 (core)",
    "telemetry 3.1.11 (core)",
    "templates 56.0.13 (core)",
    "trust 3.3.6 (core)",
    "user 3.2.7 (core)"
  ]
}

```

[1]: https://github.com/jclark-dot-org/sfcdemo/blob/master/package.json

**Comment by mdonnalley** [comment]
@jclark-dot-org Thanks for such a detailed description 🏆 

I think the problem is caused by passing in the `AuthInfo` to `Org.create` [here](https://github.com/jclark-dot-org/sfcdemo/blob/master/OrgManager.js#L56). 

`Org.create` takes [these parameters](https://github.com/forcedotcom/sfdx-core/blob/6.5.3/src/org/org.ts#L1498)

```typescript
  /**
   * Constructor Options for and Org.
   */
  export interface Options {
    aliasOrUsername?: string;
    connection?: Connection;
    aggregator?: ConfigAggregator;
    isDevHub?: boolean;
  }
``` 

Which doesn't have any overlap with `AuthInfo`. I was able to succesfully run your script by changing it to

```javascript
.then(authInfo => {console.error("authInfo: ", authInfo); return Org.create({aliasOrUsername: authInfo.username})})
```

I'm going to close the issue but please feel free to reopen it if you think I've missed something. Thanks!
